src/Esprima/Utils/AstJson.cs

```diff
@@ -0,0 +1,830 @@
+
+            public Options WithLineColumn() => WithIncludeLineColumn(true);
+            public Options WithRange()      => WithIncludeRange(true);
+            public Options WithLocation()   => WithLineColumn().WithRange();
+
+            public Options WithLocation(LocationMembersPlacement placement) =>
+                WithLocation().WithLocationMembersPlacement(placement);
+        }
+
+        public static string ToJsonString(this INode node) =>
+            ToJsonString(node, indent: null);
+
+        public static string ToJsonString(this INode node, string indent) =>
+            ToJsonString(node, Options.Default, indent);
+
+        public static string ToJsonString(this INode node, Options options) =>
+            ToJsonString(node, options, null);
+
+        public static string ToJsonString(this INode node, Options options, string indent)
+        {
+            var writer = new StringWriter();
+            WriteJson(node, writer, options, indent);
+            return writer.ToString();
+        }
+
+        public static void WriteJson(this INode node, TextWriter writer) =>
+            WriteJson(node, writer, indent: null);
+
+        public static void WriteJson(this INode node, TextWriter writer, string indent) =>
+            WriteJson(node, writer, Options.Default, indent);
+
+        public static void WriteJson(this INode node, TextWriter writer, Options options) =>
+            WriteJson(node, writer, options, null);
+
+        public static void WriteJson(this INode node, TextWriter writer, Options options, string indent)
+        {
+            if (node == null) throw new ArgumentNullException(nameof(node));
+            if (writer == null) throw new ArgumentNullException(nameof(writer));
+            if (options == null) throw new ArgumentNullException(nameof(options));
+
+            var visitor = new Visitor(new JsonTextWriter(writer, indent),
+                                      options.IncludeLineColumn, options.IncludeRange,
+                                      options.LocationMembersPlacement);
+
+            visitor.Visit(node);
+        }
+
+        public static void WriteJson(this INode node, JsonWriter writer, Options options)
+        {
+            if (node == null) throw new ArgumentNullException(nameof(node));
+            if (writer == null) throw new ArgumentNullException(nameof(writer));
+            if (options == null) throw new ArgumentNullException(nameof(options));
+
+            var visitor = new Visitor(writer,
+                                      options.IncludeLineColumn, options.IncludeRange,
+                                      options.LocationMembersPlacement);
+
+            visitor.Visit(node);
+        }
+
+        sealed class Visitor : AstVisitor
+        {
+            readonly JsonWriter _writer;
+            readonly ObservableStack<INode> _stack;
+
+            public Visitor(JsonWriter writer,
```

Comment:
Just to explain, this is not random. They are broken by group. You have the writer, the two location flags and finally their placement. I don't have a single and hard rule for laying out parameters for sake of keeping the human touch. Otherwise we might all as well throw the towel to the machines: <kbd>CTRL</kbd><kbd>K</kbd>+<kbd>D</kbd>. 😜